src/m365/planner/commands/plan/plan-details-get.ts

```diff
@@ -0,0 +1,182 @@
+import { Logger } from '../../../../cli';
+import { CommandOption} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import { PlannerPlan, PlannerPlanDetails  } from '@microsoft/microsoft-graph-types';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  planId?: string;
+  planTitle?: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerPlanDetailsGetCommand extends GraphCommand {
+  private groupId: string = "";
+
+  public get name(): string {
+    return commands.PLAN_DETAILS_GET;
+  }
+
+  public get description(): string {
+    return 'Get details of a Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planTitle = typeof args.options.planTitle !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+  
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getGroupId(args)
+      .then((groupId: string): Promise<string> => {
+        this.groupId = groupId;
+        return this.getPlanId(args);
+      })
+      .then((planId: string): Promise<PlannerPlanDetails> => {
+        args.options.planId = planId;
+        return this.getPlanDetails(args);
+      })
+      .then((res: PlannerPlanDetails): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve("");
+    }
+
+    if (args.options.ownerGroupId) {
+      return Promise.resolve(args.options.ownerGroupId);
+    }
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: [{ id: string, resourceProvisioningOptions: string[] }] }>(requestOptions)
```

Comment:
It seems like there's no reason for us to reference `resourceProvisioningOptions` here as we're not using it anywhere. Let's remove it for simplicity. This will also allow us to use the `Group` type from Graph typings, instead of defining our own type inline.